src/Components/WebAssembly/WebAssembly.Authentication/src/Models/InteractiveRequestOptions.cs

```diff
@@ -0,0 +1,192 @@
+        {
+            return false;
+        }
+        if (rawValue is JsonElement json)
+        {
+            value = Deserialize(json);
+            AdditionalRequestParameters[name] = value;
+            return true;
+        }
+        else
+        {
+            value = (TValue)rawValue;
+            return true;
+        }
+
+        [UnconditionalSuppressMessageAttribute(
+            "Trimming",
+            "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+            Justification = "All the types serialized and deserialized have annotated APIs on the parent method that ensure the members are preserved.")]
+        static TValue Deserialize(JsonElement element) => element.Deserialize<TValue>();
+    }
+
+    [UnconditionalSuppressMessage(
+        "Trimming",
+        "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+        Justification = "Serializes 'InteractiveAuthenticationRequest' into a string")]
+    internal string ToState() => JsonSerializer.Serialize(this);
+
+    [UnconditionalSuppressMessage(
+        "Trimming",
+        "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+        Justification = "Deserializes InteractiveAuthenticationRequest")]
+    internal static InteractiveRequestOptions FromState(string state) => JsonSerializer.Deserialize<InteractiveRequestOptions>(state);
+
+    internal class Converter : JsonConverter<InteractiveRequestOptions>
+    {
+        [UnconditionalSuppressMessage(
+            "Trimming",
+            "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+            Justification = "Serializes InteractiveAuthenticationRequest")]
+        public override InteractiveRequestOptions Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            var requestOptions = JsonSerializer.Deserialize<Options>(ref reader, options);
+
+            return new InteractiveRequestOptions
+            {
+                AdditionalRequestParameters = requestOptions.AdditionalRequestParameters,
+                Interaction = requestOptions.Interaction,
+                ReturnUrl = requestOptions.ReturnUrl,
+                Scopes = requestOptions.Scopes,
+            };
+        }
+
+        [UnconditionalSuppressMessage(
+            "Trimming",
+            "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+            Justification = "Serializes InteractiveAuthenticationRequest")]
```

Comment:
Not a justification

Reply:
I switched to use the serialization context